app/src/main/kotlin/com/intive/picover/parties/viewmodel/PartyDetailsViewModel.kt

```diff
@@ -1,40 +1,47 @@
 package com.intive.picover.parties.viewmodel
 
-import androidx.lifecycle.SavedStateHandle
-import androidx.lifecycle.viewModelScope
-import com.intive.picover.common.viewmodel.StatefulViewModel
-import com.intive.picover.common.viewmodel.state.ViewModelState.Error
-import com.intive.picover.common.viewmodel.state.ViewModelState.Loaded
-import com.intive.picover.common.viewmodel.state.ViewModelState.Loading
-import com.intive.picover.parties.model.Party
+import cafe.adriel.voyager.core.model.StateScreenModel
+import cafe.adriel.voyager.core.model.screenModelScope
+import cafe.adriel.voyager.hilt.ScreenModelFactory
+import com.intive.picover.common.viewmodel.state.MVIStateType
+import com.intive.picover.parties.model.PartyDetailsState
 import com.intive.picover.parties.model.toUI
 import com.intive.picover.shared.party.data.repo.PartiesRepository
-import dagger.hilt.android.lifecycle.HiltViewModel
-import javax.inject.Inject
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
 import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
 
-@HiltViewModel
-class PartyDetailsViewModel @Inject constructor(
-	savedStateHandle: SavedStateHandle,
+class PartyDetailsViewModel @AssistedInject constructor(
```

Comment:
I kept it as "ViewModel" for now to not change too much, but we can decide on naming later